### DISCUSSION TITLE
Fundo na status bar do ios

### DISCUSSION THREAD

**Discussion by brunosemfio** [Flutterando/forum#32] [Perguntas e Respostas] [answered]
A opção `SystemUiOverlayStyle.statusBarColor` funciona apenas no android. Como fariam para adicionar um fundo na status bar do iOS?

Obs.: a tela não tem uma `AppBar`.

**Comment by scduarte** [comment]
O AppBar() tem um parâmetro que vc pode tá utilizando para manipular a cor da statusbar.

**Reply by brunosemfio** [reply]
A tela em questão não tem appbar =/

Começa com uma galeria de imagens.

**Comment by levyhnunes** [comment]
Outra opção tbm pode ser o 'AnnotatedRegion':
https://stackoverflow.com/a/59672398

**Reply by brunosemfio** [reply]
O system `SystemUiOverlayStyle.statusBarColor` funciona apenas no android.

**Comment by eduardoflorence** [comment, marked as answer]
Eu já havia estudado esse problema há um tempo e a equipe do Flutter ainda não havia dado a solução (não lembro o número da issue).

A solução que eu faço é assim:
```dart
Container(
  color: Colors.green, // Cor da sua statusbar
  child: SafeArea(
    bottom: false,
    left: false
    right: false
    child: <sua galeria de imagens>
  ),
)
```
O `Container` utiliza o espaço da statusbar e o `SafeArea` protege que sua view invada a statusbar. É uma solução alternativa até que resolvam este problema.

**Reply by brunosemfio** [reply]
Na minha pergunta não deixei claro, mas quero que a galeria fique por baixo da status bar. Seu exemplo funciona, mas empurra a galeria pra baixo.

Consegui achar uma forma de fazer (que tem um cheirinho de gambiarra) mas funciona. Basta usar esse widget como um appbar:

```dart
PreferredSize(
    preferredSize: const Size.fromHeight(0.0),
    child: AppBar(
      backgroundColor: Colors.black26,
      systemOverlayStyle: SystemUiOverlayStyle.light,
    ),
  )
```

E no Scaffold precisa setar `extendBodyBehindAppBar: true`.

Com o seu exemplo:

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-09-21 at 18 23 38](https://user-images.githubusercontent.com/22257786/191613661-5b474912-cb17-4931-8c39-ede3614ae8e1.png)

Com a "gambiarra":

![Simulator Screen Shot - iPhone 14 Pro Max - 2022-09-21 at 18 21 08](https://user-images.githubusercontent.com/22257786/191614185-f5f2d6fb-a577-4d03-9b4d-e060dcc7d919.png)
